test/e2e/case42_resource_selection_test.go

```diff
@@ -0,0 +1,211 @@
+			"the policy objectSelector nor the object metadata"
+	)
+
+	// Test setup for resource selection policy tests:
+	// - FakeAPIs `case42-[1-5]-e2e`, each with a `case42: <name>` label
+	// - Deployed policy should be Compliant since the objectSelector is empty and acts as unnamed objectDefinition
+	// - Policies are patched so that the objects don't match and should be NonCompliant
+	BeforeAll(func() {
+		By("Applying prerequisites")
+		utils.Kubectl("apply", "-n", targetNs, "-f", prereqYaml)
+		DeferCleanup(func() {
+			utils.KubectlDelete("-n", targetNs, "-f", prereqYaml)
+		})
+
+		utils.Kubectl("apply", "-f", policyYaml, "-n", testNamespace)
+		DeferCleanup(func() {
+			utils.KubectlDelete("-f", policyYaml, "-n", testNamespace)
+		})
+	})
+
+	Describe("No objectSelector specified", func() {
+		It("Verifies policy is compliant with unnamed objectDefinition", func() {
+			Eventually(func() interface{} {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					policyName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetStatusMessage(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal(
+				"fakeapis [case42-1-e2e, case42-2-e2e, case42-3-e2e, case42-4-e2e, case42-5-e2e]" +
+					" found as specified in namespace " + targetNs))
+		})
+	})
+
+	DescribeTable("ObjectSelector matching all is specified", func(patch string) {
+		By("Verifying policy is noncompliant and returns no objects")
+		utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", policyName, "--type=json",
+			fmt.Sprintf(objectSelectorPatchFmt, `{"matchLabels":{"selects":"nothing"}}`),
+		)
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal(noMatchesMsg))
+
+		By("Verifying policy is compliant and returns all objects")
+		utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", policyName, "--type=json",
+			fmt.Sprintf(objectSelectorPatchFmt, "{}"),
+		)
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal(fmt.Sprintf(
+			"fakeapis [case42-1-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-2-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-3-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-4-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-5-e2e] found as specified in namespace %[1]s", targetNs)))
+	},
+		Entry("Empty selector", `{}`),
+		Entry("Empty selector", `{"matchLabels":{}}`),
+		Entry("Empty selector", `{"matchExpressions":[}`),
```

Comment:
It looks like there is a missing closing bracket. Do you know why it didn't fail?
```suggestion
		Entry("Empty selector", `{"matchExpressions":[]}`),
```

Reply:
Oh goodness. Because I never actually finished writing the test! 🤦‍♂️  The input args are never used, so it's re-running the same passing steps. 😅 I'll fix it.